File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import static java.util.stream.Collectors.*;

class Solution {
    public int solution(String[][] clothes) {
        return Arrays.stream(clothes)
                .collect(groupingBy(c -> c[1], counting())) // 종류별로 그룹화해서 개수 세기
                .values().stream()
                .reduce(1L, (ans, cnt) -> ans * (cnt + 1), (a, b) -> a * b).intValue() - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    m = {}
    
    for c in clothes:
        if c[1] in m:
            m[c[1]] += 1
        else:
            m[c[1]] = 2
    
    ans = 1
    for i in m.items():
        ans *= i[1]
    return ans - 1